public/js/clients/firefox/events.js

```diff
@@ -59,19 +65,18 @@ function resumed(_, packet) {
 function newSource(_, packet) {
   const { source } = packet;
 
-  if (NEW_SOURCE_IGNORED_URLS.indexOf(source.url) > -1) {
-    return;
-  }
-
-  if (source.introductionType == "debugger eval") {
-    return;
+  source.hidden = false;
+  if (isHidden(source)) {
+    source.url = `evals/console-evauation-${evalIndex++}.js`;
+    source.hidden = true;
```

Comment:
> I have a test for "break on next", where the debuggee pauses on a debugger eval...

Oh, I see. In the old debugger we do dynamically generate URLs for these kinds of things, just a dumb `SCRIPT#` name like `SCRIPT1`, with the number incrementing each time. I wouldn't say "console eval" because these scripts could be coming from any number of places. Chrome uses `VM#`.

I suppose we could add these to the store. Theoretically there's no reason a source needs to be in the store to display it: it could still load its source text and show it in a tab, and the `sources` state would still only have the sources in the tree. But it looks like our actions do go through the `sources` state to require it. This would mainly be an optimization.

I think I'd prefer to add `debugger eval code` and `debugger eval` sources to the store, assign them an arbitrary URL like `SOURCE#` or something, and not add another attribute. The source tree can only check the URL and work with sources that have one and it is not of the `SOURCE#` format.

The reason being I'd rather use the URL is the source for what "type" of source this is rather than adding more flags. Same goes for generated/original sources, you can detect an original source by just looked at the URL.